modules/core/src/directx.cpp

```diff
@@ -1392,11 +1393,201 @@ void convertFromD3D11Texture2D(ID3D11Texture2D* pD3D11Texture2D, OutputArray dst
         __convertFromD3D11Texture2DKHR(pD3D11Texture2D, dst);
     }
     else {
-        CV_Error(cv::Error::OpenCLApiCallError, "OpenCL: Context initilized without DirectX interoperability");
+        CV_Error(cv::Error::OpenCLApiCallError, "OpenCL: Context initialized without DirectX interoperability");
     }
 #endif
 }
 
+void getDeviceIDsByD3D11Device(ID3D11Device* pD3D11Device, std::vector<std::pair<bool, Device>>& devices) {
+    CV_UNUSED(pD3D11Device); CV_UNUSED(devices);
+#if !defined(HAVE_DIRECTX)
+    NO_DIRECTX_SUPPORT_ERROR;
+#elif !defined(HAVE_OPENCL)
+    NO_OPENCL_SUPPORT_ERROR;
+#else
+    const auto findKHRDevice =
```

Comment:
It's not clear - do we need or do not clear `devices` in this function right before fill it.
So, in this case just return device instead of passing it as argument is the solution

Reply:
Returns vector now